lib/basedriver/helpers.js

```diff
@@ -94,13 +94,20 @@ async function configureApp (app, supportedAppExtensions) {
         APPLICATIONS_CACHE.del(app);
       }
       let fileName = `appium-app${_.first(supportedAppExtensions)}`;
+
+      // to determine if we need to unzip the app, we have a number of places
+      // to look: content type, content disposition, or the file extension
+      const downloadZipName = 'appium-app.zip';
       if (headers['content-type']) {
         logger.debug(`Content-Type: ${headers['content-type']}`);
         // the filetype may not be obvious for certain urls, so check the mime type too
         if (ZIP_MIME_TYPES.includes(headers['content-type'])) {
-          fileName = 'appium-app.zip';
+          fileName = downloadZipName;
           shouldUnzipApp = true;
         }
+      } else if (ZIP_EXTS.includes(path.extname(newApp))) {
```

Comment:
not sure if it should be `else if` or just `if`

Reply:
i was thinking we'd prefer content-type if it exists

Reply:
it must be anyway very broken server, which does not return neither content type nor disposition for downloadable items

Reply:
that is github's server :-) or more likely what happens in the case of a redirect?